internal/operators/cnv/cnv_operator.go

```diff
@@ -0,0 +1,117 @@
+func NewCNVOperator(log logrus.FieldLogger) *operator {
+	return &operator{
+		log: log,
+	}
+}
+
+// GetType reports type of an operator this Operator manages
+func (o *operator) GetType() models.OperatorType {
+	return models.OperatorTypeCnv
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies() []models.OperatorType {
+	return []models.OperatorType{models.OperatorTypeLso}
+}
+
+// GetClusterValidationID returns cluster validation ID for the Operator
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDCnvRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the Operator
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDCnvRequirementsSatisfied)
+}
+
+// ValidateCluster always return "valid" result
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	// No need to validate cluster because it will be validate on per host basis
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetClusterValidationID()}, nil
+}
+
+// ValidateHost always return "valid" result
+func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) (api.ValidationResult, error) {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
+		return api.ValidationResult{Status: api.Failure}, err
+	}
+	// TODO: validate available devices on worker node like gpu and sr-iov and check whether there is enough memory to support them
+	if host.Role == models.HostRoleWorker {
+		cpu, _ := o.GetCPURequirementForWorker(ctx, cluster)
+		if inventory.CPU.Count < cpu {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient CPU to deploy CNV. Required CPU count is: " + strconv.FormatInt(inventory.CPU.Count, 10)}}, nil
+		}
+		mem, _ := o.GetMemoryRequirementForWorker(ctx, cluster)
+		if inventory.Memory.UsableBytes < mem {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient memory to deploy CNV. Required memory is: " + strconv.FormatInt(inventory.Memory.UsableBytes, 10) + " bytes"}}, nil
+		}
+	} else if host.Role == models.HostRoleMaster {
+		cpu, _ := o.GetCPURequirementForMaster(ctx, cluster)
+		if inventory.CPU.Count < cpu {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient CPU to deploy CNV. Required CPU count is: " + strconv.FormatInt(inventory.CPU.Count, 10)}}, nil
+		}
+		mem, _ := o.GetMemoryRequirementForMaster(ctx, cluster)
+		if inventory.Memory.UsableBytes < mem {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient memory to deploy CNV. Required memory is: " + strconv.FormatInt(inventory.Memory.UsableBytes, 10) + " bytes"}}, nil
```

Comment:
I wonder if it makes sense to go through all validations to capture all failures into the `Reason` field, rather than bolt at the first failure. WDYT?

Reply:
On it